ejb3/src/main/java/org/jboss/as/ejb3/cache/distributable/DistributableCacheFactoryBuilderService.java

```diff
@@ -95,6 +98,18 @@ public void installDeploymentUnitDependencies(ServiceTarget target, ServiceName
     }
 
     @Override
+    public void installDeploymentUnitDependencies(ServiceTarget target, ServiceName deploymentUnitServiceName, List<String> builderServices) {
+        for (Builder<?> builder : this.builder.getDeploymentBuilders(deploymentUnitServiceName)) {
+            if (!((ArrayList<String>)builderServices).contains(builder.getServiceName().getCanonicalName())) {
```

Comment:
@pferraro The collection of the names of the services in used here, in order to detect the duplicate ones.